cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -82,10 +82,36 @@ std::unique_ptr<table> unique(
 std::unique_ptr<table> distinct(
   table_view const& input,
   std::vector<size_type> const& keys,
+  duplicate_keep_option keep          = duplicate_keep_option::KEEP_ANY,
   null_equality nulls_equal           = null_equality::EQUAL,
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Get the row indices that denote all distinct rows in `input`.
+ *
+ * Given an `input` table_view, each row index is copied to the output array in an
+ * unspecified order if the corresponding row of `keys` columns is distinct (no other
+ * equivalent row exists in the table).
+ *
+ * Note that the returning gather map is compact: all the indices are valid.
+ *
+ * @param input The input table_view to generate gather map for distinct rows
+ * @param keys Vector of indices representing key columns from `input`
+ * @param keep Keep the first, last, any, or no rows of the found duplicates
```

Comment:
If this function only returns the indices of _distinct_ rows, why is there a _duplicate_ keep option?

Reply:
The duplicate keep option is necessary since we need to know which distinct row among duplicates is expected in the output. We can update the doc a bit if it helps:
```suggestion
 * @param keep Keep the first, last, any, or no distinct row among the duplicates
```

Reply:
If we have duplicate rows, the index of the first/last/none duplicate row will be returned.

This is necessary to gather the input rows using the returning indices (as gather map) so we will have the result of `cudf::distinct` with a `keep` option.

Reply:
Then the docs need to be updated because that is not what it says it does:

> @brief Get the row indices that denote all distinct rows in `input`.

This tells me it returns the indices of rows that are _distinct_ as in, do not have any duplicates, or the equivalent of `KEEP_NONE`.

Reply:
You need to find a way to say that the rows referenced by the returned indices will all be distinct from each other.